src/Compilers/CSharp/Portable/Compilation/AttributeSemanticModel.cs

```diff
@@ -40,8 +40,13 @@ internal sealed class AttributeSemanticModel : MemberSemanticModel
         /// </summary>
         public static AttributeSemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, AttributeSyntax syntax, NamedTypeSymbol attributeType, AliasSymbol aliasOpt, Symbol? attributeTarget, Binder rootBinder, ImmutableDictionary<Symbol, Symbol> parentRemappedSymbolsOpt)
         {
+            // PROTOTYPE: There was a conflict here that I resolved "naively". It needs to be revised more carefully, and will need more tests.
```

Comment:
funny! I guess we ended up having and solving a similar problem for `nameof(param)`. Maybe we will end up adjusting the implementation in this PR to lean on what was introduced for `nameof(param)`. cc @jcouv.

Reply:
Yeah I'm planning to take a deeper look at the changes done in https://github.com/dotnet/roslyn/commit/8fa902f3859be74c449d1687c88fe98cf5834b45 and have more IOp tests. I'm also waiting on https://github.com/dotnet/roslyn/issues/61492 (I'll try to take a look at it too). I don't know if I'll get to that soon or not.

Reply:
Just sharing a bit of background if that could help:

More identifiers are visible inside certain `nameof` operators than right outside the `nameof`. The way this was previously done was to put all the things in scope (ie. add binders for type parameters and parameters in the chain) but then filter them during lookup. That filtering causes problems:
1. it is different for general lookup (`BindIdentifier`) vs. type lookup (`BindType`)
2. it's broken on local functions (filter is conditioned on being "inside a method", `IsInMethodBody`, which worked before we could have multiple nested scopes of method parameters and type parameters)
3.  it's difficult to replicate properly in semantic and speculative model (if I recall, this has to do with determining whether `nameof(x)` might be an invocation or a `nameof` operator)

The way this same problem is solved for "extended nameof scope" feature is that the binder for `nameof` brings relevant parameters and type parameters into scope (conditionally). That binder can tell whether the `nameof` is inside an attribute and whether the `nameof` is an invocation or an operator.

This new model avoids the problems described above. I tried converting existing scenarios to use this new model, but doing so without introducing meaningful compat breaks turned out badly. I don't remember which scenario was the last straw.